File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        for(int i=m,j=0;i<nums1.length && j<n;i++,j++) {
            nums1[i] = nums2[j];
        }
        
        for(int i=1;i<nums1.length;i++) {
            int picked = nums1[i];
            int j=i-1;
            while(j>=0 && nums1[j]>picked) {
                nums1[j+1] = nums1[j];
                j--;
            }
            nums1[j+1] = picked;
        }
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        l1 = m - 1
        l2 = n - 1
        i = m+n - 1
        while(l2 >= 0):
            if l1 >= 0 and nums1[l1] > nums2[l2]:
                nums1[i] = nums1[l1]
                i-=1
                l1-=1
            else:
                nums1[i] = nums2[l2]
                i-=1
                l2-=1
        return nums1
